ocis-pkg/sync/mutex.go

```diff
@@ -0,0 +1,74 @@
+package sync
+
+import (
+	"sync"
+)
+
+// NRWMutex works the same as RWMutex, the only difference is that it stores mutexes in a map and reuses them.
+// It's handy if you want to write-lock, write-unlock, read-lock and read-unlock for specific names only.
+type NRWMutex struct {
+	m  sync.Mutex
+	mm map[string]*nrw
+}
+
+type nrw struct {
+	m sync.RWMutex
+	c int
+}
+
+// NewNRWMutex returns a new instance of NRWMutex.
+func NewNRWMutex() NRWMutex {
+	return NRWMutex{mm: make(map[string]*nrw)}
+}
+
+// Lock locks rw for writing.
+func (c *NRWMutex) Lock(k string) {
+	c.m.Lock()
+	m := c.get(k)
+	m.c++
```

Comment:
isn't `c` here acting like a `sync.WaitGroup`? 🤔

Reply:
yes, but i find that it's easier to read and maintain by using mutex